ipam/allocator.go

```diff
@@ -66,18 +67,22 @@ type Allocator struct {
 	isKnownPeer       func(mesh.PeerName) bool
 	quorum            func() uint
 	now               func() time.Time
+	isCIDRAligned     bool            // should donate only CIDR aligned ranges
+	monitor           monitor.Monitor // monitor tracks changes in address ranges owned by us
 }
 
 type Config struct {
-	OurName     mesh.PeerName
-	OurUID      mesh.PeerUID
-	OurNickname string
-	Seed        []mesh.PeerName
-	Universe    address.Range
-	IsObserver  bool
-	Quorum      func() uint
-	Db          db.DB
-	IsKnownPeer func(name mesh.PeerName) bool
+	OurName       mesh.PeerName
+	OurUID        mesh.PeerUID
+	OurNickname   string
+	Seed          []mesh.PeerName
+	Universe      address.Range
+	IsObserver    bool
+	Quorum        func() uint
+	Db            db.DB
+	IsKnownPeer   func(name mesh.PeerName) bool
+	IsCIDRAligned bool
```

Comment:
I suggest this is made non-optional.
It will make the allocation of ranges less fair in many cases, but that doesn't seem like a big problem, and it will reduce the code bloat.